components/logins/src/store.rs

```diff
@@ -297,15 +312,29 @@ impl LoginStore {
         Ok(())
     }
 
-    #[handle_error(Error)]
-    pub fn shutdown(&self) -> ApiResult<()> {
-        if let Some(conn) = self.db.lock().take() {
-            conn.shutdown()
+    fn shutdown_db(&self) -> Result<()> {
+        if let Some(db) = self.db.lock().take() {
+            db.shutdown()
         } else {
             Ok(())
         }
     }
 
+    fn shutdown_encdec(&self) -> Result<()> {
+        self.encdec
+            .lock()
+            .take()
+            .ok_or(Error::EncryptorDecryptorClosed)?;
```

Comment:
I don't think this needs to be an error.  If `shutdown()` is called twice, that seems okay.  It's a close call though, however you decide is fine with me.

Reply:
I agree, good point!